components/api/filter/validator-filter.bal

```diff
@@ -35,14 +35,22 @@ public type validateRequestFilter object {
             request.removeHeader(constants:AUTHENTICATED_USER);
         }
         string token = "";
-        if(request.hasHeader(constants:AUTHORIZATION_HEADER)) {
+        if(request.hasHeader(constants:AUTHORIZATION_HEADER) && request.hasHeader(constants:COOKIE_HEADER)) {
             string tokenHeaderValue = request.getHeader(constants:AUTHORIZATION_HEADER);
             string[] splittedToken = tokenHeaderValue.split(" ");
             if splittedToken.length() != 2 || !(splittedToken[0] == "Bearer" || splittedToken[0] == "bearer") {
-                log:printError("Not not receive the token in proper format");
+                log:printError("Did not receive the token in proper format");
                 return true;
             }
-            token = splittedToken[1];
+            string tokenLast = splittedToken[1];
+            string cookieHeader = request.getHeader(constants:COOKIE_HEADER);
+            string|error tokenFirst = getCookie(cookieHeader);
+            if tokenFirst is error {
+                log:printError("Cookie value could not be resolved. Passing to the next filter", err = tokenFirst);
+                return true;
+            } else{
+                token = io:sprintf("%s%s", tokenFirst, tokenLast);
+            }
             if "" == token {
```

Comment:
Token is assigned in line #52, and hence check whether it's empty in that else section it self, rather having another if block.